### ISSUE TITLE
Observation: example cifar_train failes with assertions enabled

### ISSUE THREAD

**Issue by Bparvez** [tiny-dnn/tiny-dnn#842]
Tested on: 8c5d99 , on a ubuntu 16.04 LTS Virtual Machine.

Compiled using: 
`g++ -g -O0 -std=c++14 -I /home/jevois/tiny-dnn-myown/ -L /usr/local/lib -Wall -o model_train.out model_train.cpp -lprotobuf -lopencv_core -lopencv_highgui -lopencv_imgproc -lpthread`

worked when disabling assertion by compiling using:
`g++ -g -O0 -std=c++14 -I /home/jevois/tiny-dnn-myown/ -L /usr/local/lib -Wall -o model_train.out model_train.cpp -lprotobuf -lopencv_core -lopencv_highgui -lopencv_imgproc -lpthread -DNDEBUG`

Error output:

>➜  dnn_test_project ./model_train.out --data_path cifar-10-batches-bin --learning_rate 0.01 --epochs 30 --minibatch_size 10 --backend_type internal                                                       
Running with the following parameters:
Data path: cifar-10-batches-bin
Learning rate: 0.01
Minibatch size: 10
Number of epochs: 30
Backend type: Internal

>load models...
start learning

>0%   10   20   30   40   50   60   70   80   90   100%
|----|----|----|----|----|----|----|----|----|----|
model_train.out: /home/jevois/tiny-dnn-myown/tiny_dnn/layers/layer.h:241: void tiny_dnn::layer::set_in_data(const std::vector<const std::vector<float, tiny_dnn::aligned_allocator<float, 64ul> >*>*, size_t): Assertion `dst_data[j].size() == src_data[j]->size()' failed.
[1]    29303 abort (core dumped)  ./model_train.out --data_path cifar-10-batches-bin --learning_rate 0.01  30  

I am new to tiny-dnn and haven't looked into the inner workings to say how or if this effects anything. Just pointing it out.

**Comment by nyanp** [comment]
@Bparvez 
Thanks for your reporting, this was a bug introduced by a44a52e, and fixed by #853 :)
